docs/terminal/advanced.md

```diff
@@ -0,0 +1,182 @@
+
+### Custom sequence keybindings
+
+The `workbench.action.terminal.sendSequence` command can be used to send a specific sequence of text to the terminal, this includes escape sequences that are interpreted specially by the shell. This enables things like sending arrow keys, enter, cursor moves, etc. For example, the below sequence jumps over the word to the left of the cursor (`kbstyle(Ctrl+Left)`) and presses backspace:
+
+```jsonc
+{
+  "key": "ctrl+u",
+  "command": "workbench.action.terminal.sendSequence",
+  "args": {
+    "text": "\u001b[1;5D\u007f"
+  }
+}
+```
+
+This feature supports [variable substitution](/docs/editor/variables-reference.md).
+
+Note that the command only works with the `\u0000` format for using characters via their character code (not `\x00`). Read more about these hex codes and the sequences terminals work with on the following resources:
+
+* [XTerm Control Sequences](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html)
+* [List of C0 and C1 control codes](https://github.com/xtermjs/xterm.js/blob/0e45909c7e79c83452493d2cd46d99c0a0bb585f/src/common/data/EscapeSequences.ts)
+
+## Confirmation dialogs
+
+In order to avoid additional noise and user actions, the terminal does not show warning dialogs when exiting processes. If warnings are desirable they can be configured with the following settings:
+
+- `terminal.integrated.confirmOnExit` - Controls whether to confirm when the window closes if there are active debug sessions.
+- `terminal.integrated.confirmOnKill` - Controls whether to confirm killing terminals when they have child processes.
+- `terminal.integrated.showExitAlert` - Controls whether to show the alert "The terminal process terminated with exit code" when exit code is non-zero.
+
+## Auto replies
+
+The terminal is capable of providing an input response to the shell automatically if an exact sequence of output is received. This could be used for all sorts of things but the most common use case is to automatically reply to the quirk when hitting `kbstyle(Ctrl+C)` in batch scripts that asks whether the user wants to terminate the batch job. To automatically dismiss this message, add this setting:
+
+```json
+{
+  "terminal.integrated.autoReplies": {
+    "Terminate batch job (Y/N)": "Y\r"
+  }
+}
+```
+
+Notice the `\r` here which means `kbstyle(Enter)`, much like [custom sequence keybindings](#_custom-sequence-keybindings) this supports sending escape sequences to the shell.
+
+No auto replies are configured by default as this deals with shell input which is sensitive.
+
+## Unicode and emoji support
+
+The terminal has both unicode and emoji support, since these are being used in a terminal environment there are some caveats to that support:
```

Comment:
```suggestion
The terminal has both unicode and emoji support; since these are being used in a terminal environment there are some caveats to that support:
```

Reply:
I think `,` might be more correct here? @gregvanl thoughts?

Reply:
they're both independent clauses, so a semi-colon is correct here

Reply:
I think I would break it up into two sentences because the second part is a lead to the caveats.
"The terminal has both Unicode and emoji support. When these characters are used in the terminal, there are some caveats to that support:"